src/autoscan.cc

```diff
@@ -174,30 +168,29 @@ int AutoscanList::removeByObjectID(int objectID)
 {
     AutoLock lock(mutex);
 
-    for (size_t i = 0; i < list.size(); i++) {
-        if (objectID == list[i]->getObjectID()) {
-            auto dir = list[i];
-            dir->setScanID(INVALID_SCAN_ID);
-            list.erase(list.begin() + i);
-            return i;
-        }
+    auto it = std::find_if(list.begin(), list.end(), [=](const auto& item) { return objectID == item->getObjectID(); });
+    if (it == list.end()) {
+        return INVALID_SCAN_ID;
     }
-    return INVALID_SCAN_ID;
+
+    auto dir = *it;
+    dir->setScanID(INVALID_SCAN_ID);
+    list.erase(list.begin() + std::distance(list.begin(), it));
```

Comment:
This is a really strange math... Shouldn't be just `list.erase(it);`?
Also what is the type of `list`? because `erase` on `vector` invalidates iterators (in any case `it` should point at deleted element). If return value is used to preserve the original behavior you need to compute distance first and then return. If it's a `bool` in disguise just return `true`/`false`.

I would also probably just do `(*it)->setScanID(INVALID_SCAN_ID)` without creating a temporary.

Reply:
I think I adjusted it right.